dbms/src/Storages/MergeTree/ReplicatedMergeTreeQuorumWriter.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include <unordered_map>
+#include <IO/ReadBuffer.h>
+#include <IO/ReadBufferFromString.h>
+#include <IO/WriteBuffer.h>
+#include <IO/WriteBufferFromString.h>
+#include <IO/ReadHelpers.h>
+
+#include <Storages/MergeTree/MergeTreeDataPart.h>
+
+namespace DB
+{
+
+struct ReplicatedMergeTreeQuorumWriter
```

Comment:
I don't like the -Writer name. This is a struct so it should be a "dumb data item" and contain as little logic as possible. But the Writer name hints at a class with some complicated internal logic. But this struct should just represent last parts written by quorum inserts and should be named accordingly.